test/v2/client/test_client.go

```diff
@@ -518,67 +558,138 @@ func (c *TestClient) ReadBlobFromRelays(
 
 	for _, relayID := range relayKeys {
-		start := time.Now()
-
-		c.logger.Debugf("Reading blob from relay %d", relayID)
-		blobBytesFromRelay, err := c.relayClient.GetBlob(ctx, relayID, key)
+		err := c.ReadBlobFromRelay(ctx, key, relayID, expectedPayload, blobLengthSymbols, timeout)
 		if err != nil {
-			return fmt.Errorf("failed to read blob from relay: %w", err)
+			return fmt.Errorf("failed to read blob from relay %d: %w", relayID, err)
 		}
+	}
 
-		c.metrics.reportRelayReadTime(time.Since(start), relayID)
+	return nil
+}
 
-		blob, err := coretypes.DeserializeBlob(blobBytesFromRelay, blobLengthSymbols)
-		if err != nil {
-			return fmt.Errorf("failed to deserialize blob: %w", err)
-		}
+// ReadBlobFromRelay reads a blob from the relay and compares it to the given payload.
+func (c *TestClient) ReadBlobFromRelay(
+	ctx context.Context,
+	key corev2.BlobKey,
+	relayKey corev2.RelayKey,
+	expectedPayload []byte,
+	blobLengthSymbols uint32,
+	timeout time.Duration,
+) error {
 
-		payload, err := blob.ToPayload(c.payloadClientConfig.PayloadPolynomialForm)
-		if err != nil {
-			return fmt.Errorf("failed to decode blob: %w", err)
-		}
+	start := time.Now()
 
-		payloadBytesFromRelay := payload.Serialize()
+	if timeout > 0 {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, timeout)
+		defer cancel()
+	}
 
-		if !bytes.Equal(payloadBytesFromRelay, expectedPayload) {
-			return fmt.Errorf("payloads do not match")
-		}
+	blobBytesFromRelay, err := c.relayClient.GetBlob(ctx, relayKey, key)
+	if err != nil {
+		return fmt.Errorf("failed to read blob from relay: %w", err)
+	}
+
+	c.metrics.reportRelayReadTime(time.Since(start), relayKey)
+
+	blob, err := coretypes.DeserializeBlob(blobBytesFromRelay, blobLengthSymbols)
+	if err != nil {
+		return fmt.Errorf("failed to deserialize blob: %w", err)
+	}
+
+	payload, err := blob.ToPayload(c.payloadClientConfig.PayloadPolynomialForm)
+	if err != nil {
+		return fmt.Errorf("failed to decode blob: %w", err)
+	}
+
+	payloadBytesFromRelay := payload.Serialize()
+
+	if !bytes.Equal(payloadBytesFromRelay, expectedPayload) {
+		return fmt.Errorf("payloads do not match")
 	}
 
 	return nil
 }
 
+// GetCurrentBlockNumber retrieves the current block number from the indexed chain state.
+func (c *TestClient) GetCurrentBlockNumber(ctx context.Context) (uint64, error) {
+	currentBlockNumber, err := c.indexedChainState.GetCurrentBlockNumber(ctx)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get current block number: %w", err)
+	}
+	return uint64(currentBlockNumber), nil
+}
+
 // ReadBlobFromValidators reads a blob from the validators and compares it to the given payload.
 func (c *TestClient) ReadBlobFromValidators(
 	ctx context.Context,
 	blobKey corev2.BlobKey,
 	blobVersion corev2.BlobVersion,
 	blobCommitments encoding.BlobCommitments,
 	quorums []core.QuorumID,
-	expectedPayloadBytes []byte) error {
+	expectedPayloadBytes []byte,
+	timeout time.Duration,
```

Comment:
consider writing a doc for this `timeout`, too

Reply:
```
// The timeout provided is a timeout for each read from a quorum, not all reads as a whole.
```